<a href="https://colab.research.google.com/github/Angy-face/Esan-Coding-SS2/blob/main/Diamonds_Dataset.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [10]:
!pip install -U pandas-profiling

  Preparing metadata (setup.py) ... done
   ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 324.4/324.4 kB 20.6 MB/s eta 0:00:00
   ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 390.9/390.9 kB 25.5 MB/s eta 0:00:00
   ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 296.5/296.5 kB 23.7 MB/s eta 0:00:00
   ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 686.1/686.1 kB 39.8 MB/s eta 0:00:00
   ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 104.8/104.8 kB 9.2 MB/s eta 0:00:00
   ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 4.5/4.5 MB 87.6 MB/s eta 0:00:00
  Created wheel for htmlmin: filename=htmlmin-0.1.12-py3-none-any.whl size=27081 sha256=a3f96df29ad7a5bf8f93883b8b44e12dfa2a83cd671181ea6de6c822dd7815d3
  Stored in directory: /root/.cache/pip/wheels/dd/91/29/a79cecb328d01739e64017b6fb9a1ab9d8cb1853098ec5966d
Successfully built htmlmin


In [12]:
import kagglehub
import pandas as pd
import shutil as st
import matplotlib.pyplot as plt
import seaborn as sns
from ydata_profiling import ProfileReport

# 1.Data retrieving

In [2]:
path = kagglehub.dataset_download("shivam2503/diamonds")
st.move(path,"/content/Data")

100%|██████████| 733k/733k [00:00<00:00, 23.5MB/s]

Extracting files...


'/content/Data'

In [9]:
df  = pd.read_csv("/content/Data/diamonds.csv",index_col=0)
df

carat        cut color clarity  depth  table  price     x     y     z
1       0.23      Ideal     E     SI2   61.5   55.0    326  3.95  3.98  2.43
2       0.21    Premium     E     SI1   59.8   61.0    326  3.89  3.84  2.31
3       0.23       Good     E     VS1   56.9   65.0    327  4.05  4.07  2.31
4       0.29    Premium     I     VS2   62.4   58.0    334  4.20  4.23  2.63
5       0.31       Good     J     SI2   63.3   58.0    335  4.34  4.35  2.75
...      ...        ...   ...     ...    ...    ...    ...   ...   ...   ...
53936   0.72      Ideal     D     SI1   60.8   57.0   2757  5.75  5.76  3.50
53937   0.72       Good     D     SI1   63.1   55.0   2757  5.69  5.75  3.61
53938   0.70  Very Good     D     SI1   62.8   60.0   2757  5.66  5.68  3.56
53939   0.86    Premium     H     SI2   61.0   58.0   2757  6.15  6.12  3.74
53940   0.75      Ideal     D     SI2   62.2   55.0   2757  5.83  5.87  3.64

[53940 rows x 10 columns]

## 💎 Diamond Attributes 💎

* **Price** ▶ \$326--\$18,823
* **Carat** ▶ 0.2--5.01
* **Cut** ▶ Fair, Good, Very Good, Premium, Ideal
* **Color** ▶ J(worst)--D(best)
* **Clarity** ▶ I1 (worst), SI2, SI1, VS2, VS1, VVS2, VVS1, IF (best)
* **Length (x)** ▶ 0--10.74 mm
* **Width (y)** ▶ 0--58.9 mm
* **Depth (z)** ▶ 0--31.8 mm
* **Depth Percentage** ▶ 43--79%
* **Table** ▶ 43--95%

# 2.Data cleaning

In [18]:
# profile = ProfileReport(df)
# profile
df.describe()

carat         depth         table         price             x  \
count  53940.000000  53940.000000  53940.000000  53940.000000  53940.000000   
mean       0.797940     61.749405     57.457184   3932.799722      5.731157   
std        0.474011      1.432621      2.234491   3989.439738      1.121761   
min        0.200000     43.000000     43.000000    326.000000      0.000000   
25%        0.400000     61.000000     56.000000    950.000000      4.710000   
50%        0.700000     61.800000     57.000000   2401.000000      5.700000   
75%        1.040000     62.500000     59.000000   5324.250000      6.540000   
max        5.010000     79.000000     95.000000  18823.000000     10.740000   

                  y             z  
count  53940.000000  53940.000000  
mean       5.734526      3.538734  
std        1.142135      0.705699  
min        0.000000      0.000000  
25%        4.720000      2.910000  
50%        5.710000      3.530000  
75%        6.540000      4.040000  
max       58.900000     31.800000

In [17]:
df.info()

<class 'pandas.core.frame.DataFrame'>
Index: 53940 entries, 1 to 53940
Data columns (total 10 columns):
 #   Column   Non-Null Count  Dtype  
---  ------   --------------  -----  
 0   carat    53940 non-null  float64
 1   cut      53940 non-null  object 
 2   color    53940 non-null  object 
 3   clarity  53940 non-null  object 
 4   depth    53940 non-null  float64
 5   table    53940 non-null  float64
 6   price    53940 non-null  int64  
 7   x        53940 non-null  float64
 8   y        53940 non-null  float64
 9   z        53940 non-null  float64
dtypes: float64(6), int64(1), object(3)
memory usage: 6.5+ MB


In [19]:
df.isna().sum()

carat      0
cut        0
color      0
clarity    0
depth      0
table      0
price      0
x          0
y          0
z          0
dtype: int64

In [20]:
df.sample(10)

carat      cut color clarity  depth  table  price     x     y     z
6595    0.90  Premium     G     VS2   60.6   61.0   4079  6.19  6.13  3.73
24863   2.50     Fair     H     SI2   64.9   58.0  13278  8.46  8.43  5.48
43267   0.30    Ideal     I     SI1   61.3   56.0    506  4.33  4.32  2.65
30924   0.34  Premium     E     VS2   61.9   58.0    745  4.43  4.45  2.75
40230   0.40    Ideal     G    VVS1   61.2   56.0   1123  4.75  4.79  2.92
39782   0.36    Ideal     D     VS2   61.6   56.0   1094  4.57  4.55  2.81
23948   1.56    Ideal     H     SI1   62.2   55.0  12082  7.41  7.45  4.62
23891   2.00    Ideal     H     SI2   59.7   57.0  11988  8.29  8.22  4.93
43953   0.31     Good     F     SI1   63.5   56.0    516  4.30  4.33  2.74
46543   0.51  Premium     E     VS2   61.0   60.0   1781  5.17  5.12  3.14

#3.Data Visualization // Data Analysis

In [26]:
correlation = df.corr(method='pearson', numeric_only=True)

carat     depth     table     price         x         y         z
carat  1.000000  0.028224  0.181618  0.921591  0.975094  0.951722  0.953387
depth  0.028224  1.000000 -0.295779 -0.010647 -0.025289 -0.029341  0.094924
table  0.181618 -0.295779  1.000000  0.127134  0.195344  0.183760  0.150929
price  0.921591 -0.010647  0.127134  1.000000  0.884435  0.865421  0.861249
x      0.975094 -0.025289  0.195344  0.884435  1.000000  0.974701  0.970772
y      0.951722 -0.029341  0.183760  0.865421  0.974701  1.000000  0.952006
z      0.953387  0.094924  0.150929  0.861249  0.970772  0.952006  1.000000